### PR TITLE
Add a missing check on s->s3->tmp.pkey

### LINKED ISSUE #7370: TLS1_2: fatal decrypt_error
We have a mips32 based embedded device with openssl 1.0.2o .
The remote openssl tls1_2 connect to the webserver installed on this device is failing with "fatal decrypt_error" . 
Connection works with tls1_1. The local (on same device) connect also works in both cases i.e. tls1_2 and tls1_1.

Openssl Version - 1.0.2o and 1.0.2p
Target - mips32 (compiler gcc)

Tried openssl config with various config options (Compiler/OS - linux-generic32, linux-mips32,  Flags- no-asm, disabling optimization). But facing same error in all these cases.

`#openssl s_client -connect  <ipaddr>:443 -tls1_2 -msg`

```
----------------
>>> TLS 1.2 Alert [length 0002], fatal decrypt_error
    02 33
140463749879456:error:04091068:rsa routines:INT_RSA_VERIFY:bad signature:rsa_sign.c:291:
140463749879456:error:1408D07B:SSL routines:SSL3_GET
----------------
---
No client certificate CA names sent
SSL handshake has read 1305 bytes and written 7 bytes
---
New, (NONE), Cipher is (NONE)
Server public key is 2048 bit
Secure Renegotiation IS supported
Compression: NONE
Expansion: NONE
SSL-Session:
    Protocol  : TLSv1.2
    Cipher    : 0000
    Session-ID: 
    Session-ID-ctx: 
    Master-Key: 
    Key-Arg   : None
    PSK identity: None
    PSK identity hint: None
    SRP username: None
    Start Time: 1539041589
    Timeout   : 7200 (sec)
    Verify return code: 18 (self signed certificate)

```

### LINKED ISSUE #7374: NULL Dereference - Missing check on s->s3->tmp.pkey
s->s3->tmp.pkey is checked here:

https://github.com/openssl/openssl/blob/master/ssl/statem/statem_srvr.c#L3147

skey = s->s3->tmp.pkey;
    if (skey == NULL) { ...

But it is not checked here:

https://github.com/openssl/openssl/blob/master/ssl/statem/statem_srvr.c#L3202

EVP_PKEY *skey = s->s3->tmp.pkey;

No check on skey and if skey is NULL, skey is dereferenced (from->type) in EVP_PKEY_copy_parameters 

https://github.com/openssl/openssl/blob/master/crypto/evp/p_lib.c#L75

Thank you!

### PR DESCRIPTION
**Pull request by ManSoSec**
This should fix #7374

### CODE REVIEW

**Comment by ManSoSec** [comment]
> Please can you submit a CLA

Would you please let me know which email I can send my CLA? I couldn't find any email in the website regarding this issue.

Thank you!

File: ssl/statem/statem_srvr.c [hunk 1]
```diff
@@ -3224,6 +3224,12 @@ static int tls_process_cke_ecdhe(SSL *s, PACKET *pkt)
                      SSL_R_LENGTH_MISMATCH);
             goto err;
         }
+        if (skey == NULL) {
+            SSLfatal(s, SSL_AD_INTERNAL_ERROR, SSL_F_TLS_PROCESS_CKE_ECDHE,
+                     SSL_R_MISSING_TMP_DH_KEY);
```
[the remark this answers is not in the document]
Comment (ManSoSec) [reply ssl/statem/statem_srvr.c]: Modified it!

**Comment by mattcaswell** [comment]
> Would you please let me know which email I can send my CLA? I couldn't find any email in the website regarding this issue.

Its in the CLA itself. But don't worry, I have it anyway and have updated our records.

Close/reopen to kick CLA bot

**Comment by mattcaswell** [comment]
Pushed to master and 1.1.1. Thanks.

**Pull request closed without merging**